### PR TITLE
chore: adds reactJS

### PR DESCRIPTION
**Pull request by caminoezequiel**
Link to Test: https://play.decentraland.zone/branch/chore/adds_react/index.html?ENV=zone

### CODE REVIEW

**Review by pravusjif — APPROVED** [review]
Just tested the build, and had no issues.

File: kernel/Makefile
Comment (menduz) [inline kernel/Makefile:271]: is this still valid?

Comment (caminoezequiel) [reply kernel/Makefile:271]: @menduz Yes, it is. Watching website.json allows us see the updates of kernel in reactjs when we are in local

File: kernel/packages/shared/Html.ts
Comment (pablitar) [inline kernel/packages/shared/Html.ts]: Love having all this nasty stuff in the same place!

**Review by menduz — APPROVED** [review]
Looks good, resolve the conflicts and ship it :shipit:

File: website/src/components/auth/LoginHeader.tsx
Comment (pablitar) [inline website/src/components/auth/LoginHeader.tsx]:
```suggestion
      Inmerse yourself into the first virtual world fully owned by its users.
```

File: kernel/packages/shared/dao/sagas.ts
Comment (pablitar) [inline kernel/packages/shared/dao/sagas.ts:51]: I suppose we need this to be able to run the avatar editor, right? isn't there another workaround?

File: kernel/packages/shared/ethereum/Web3Connector.ts
Comment (pablitar) [inline kernel/packages/shared/ethereum/Web3Connector.ts:9]: This is really nice

File: kernel/packages/shared/ethereum/provider.ts
Comment (pablitar) [inline kernel/packages/shared/ethereum/provider.ts:1]: Love all the deleted lines here

File: kernel/packages/shared/index.ts
Comment (pablitar) [inline kernel/packages/shared/index.ts:17]: does this mean that now initShared can be called multiple times?

File: kernel/packages/shared/profiles/sagas.ts
Comment (pablitar) [inline kernel/packages/shared/profiles/sagas.ts:325]: We already talked about this, but ensure to use saveProfileRequest so we minimize alternative flows!

File: kernel/packages/shared/session/sagas.ts
Comment (pablitar) [inline kernel/packages/shared/session/sagas.ts:96]: Nice

File: kernel/packages/shared/session/sagas.ts
Comment (pablitar) [inline kernel/packages/shared/session/sagas.ts]: Just a sanity check: `yield loginCompleted` awaited for the background login to finish... I think.

This completely changes that behaviour to marking the login completed when this line is hit.

Is this the expected behaviour?

We have a couple of `loginCompleted.resolve()` above

File: kernel/packages/unity-interface/BrowserInterface.ts
Comment (pablitar) [inline kernel/packages/unity-interface/BrowserInterface.ts:217]: Can't you put this in the `HTML` module that you've created?

**Review by pablitar — APPROVED** [review]
This is huge. Great work!

Please review the comments before merging, but no need to block the PR because of them.

We may have to fix some things, but once this thing is in master, we can work much faster an in parallel to make it stable :)

File: kernel/packages/unity-interface/BrowserInterface.ts
Comment (caminoezequiel) [reply kernel/packages/unity-interface/BrowserInterface.ts:217]: yes! of course, I've seen that line duplicated in so many places

File: kernel/packages/shared/dao/sagas.ts
Comment (caminoezequiel) [reply kernel/packages/shared/dao/sagas.ts:51]: It's necessary to kwon if a profile already exists or not. Today the realm is initialized after calling `loginCompleted` but now we need to kwon if the address has a profile before `loginCompleted` to kwon if user need to continue with the sign up or sing in flow.

File: kernel/packages/shared/index.ts
Comment (caminoezequiel) [reply kernel/packages/shared/index.ts:17]: No, it is called only one time, the difference is that in react app we need to have the redux store before `initShared` is called then here we check it.

File: kernel/packages/shared/session/sagas.ts
Comment (caminoezequiel) [reply kernel/packages/shared/session/sagas.ts]: Yes, i think so. Let me check why it changed and i'll talk you later

File: .circleci/config.yml
Comment (menduz) [inline .circleci/config.yml]: This won't work in .today or .org

File: kernel/packages/build-ecs/index.ts
Comment (menduz) [inline kernel/packages/build-ecs/index.ts]: Please restore this, it is **_REALLY_** necessary

File: kernel/packages/shared/ethereum/connector/ConnectorFactory.ts
Comment (menduz) [inline kernel/packages/shared/ethereum/connector/ConnectorFactory.ts:8]: this class can be a simple function

File: kernel/packages/shared/session/actions.ts
Comment (menduz) [inline kernel/packages/shared/session/actions.ts]: missing [] in the action name

File: kernel/packages/build-ecs/index.ts
Comment (caminoezequiel) [reply kernel/packages/build-ecs/index.ts]: yes!, of course. I don't know why these lines are comment, maybe @2fd knows

File: .circleci/config.yml
Comment (caminoezequiel) [reply .circleci/config.yml]: @menduz Do you know how we can set the top level domain dynamically? can we know it from the branch name?

Comment (menduz) [reply .circleci/config.yml]: we have no way to do it at any build stage of building currently, as they are branch agnostic... is there any chance to not require the public_url to generate the artifacts?

Comment (caminoezequiel) [reply .circleci/config.yml]: it's require only for branches because we are deploying on a sub directory so we can use absolute paths. It's the reason why I ignore `master` and `tags` on lines 301 to 303

File: website/src/components/auth/EthLogin.stories.tsx
Comment (BrianAmadori) [inline website/src/components/auth/EthLogin.stories.tsx]:
```suggestion
export const LoginSigning = Template.bind({});
LoginSigning.args = {
```

File: kernel/packages/shared/session/types.ts
Comment (BrianAmadori) [inline kernel/packages/shared/session/types.ts]:
```suggestion
  SIGN_IN = 'signIn',
  SIGN_UP = 'signUp',
  CONNECT_ADVICE = 'connect_advice',
  SIGN_ADVICE = 'sign_advice',
```

**Review by pravusjif — APPROVED** [review]
awsome work man!

**Pull request merged**
